lib/Core/EpisodeAction/EpisodeActionReader.php

```diff
@@ -4,41 +4,36 @@
 namespace OCA\GPodderSync\Core\EpisodeAction;
 
 class EpisodeActionReader {
+
+    const EPISODEACTION_IDENTIFIER = 'EpisodeAction{';
+
 	/**
-	 * Reads and parses an EpisodeActions string and returns an EpisodeAction array
-	 *
 	 * @param string $episodeActionString
-	 * @return array
+	 * @return EpisodeAction[]
 	 */
 	public function fromString(string $episodeActionString): array {
 		
-		$episodeActions = array();
+		$episodeActions = [];
 
-		$seek = 0;
+        $episodeActionStrings = explode(self::EPISODEACTION_IDENTIFIER, $episodeActionString);
 
-        while (strpos($episodeActionString, 'EpisodeAction{', $seek) >= $seek) {
-            if (($seek = strpos($episodeActionString, 'EpisodeAction{', $seek)) === false) {
-                continue;
-            }
+        for($i = 1; $i < count($episodeActionStrings); $i++) {
```

Comment:
Please use a for each loop here. One only needs a for loop  when the programming learning course asks one to use one ;-)

Reply:
Okay, I only did this because the first array element needs to be skipped. Will try to do this without for, maybe with `strlen()` and `continue`, with a separate counter or with `array_slice()`. What do you think would be best?

Reply:
Sorry i missed that. You can remove the first element of the array with `array_shift($episodeActionStrings);`

Nevertheless this parsing logic (regex) was hard to read/understand before thus hard to extend like we experience now. Lets just make it work with a foreach loop and then later find a more readable/maintainable parsing approach

Reply:
Yes, if this wouldn't change the logic too much I think it would be best if, in long-term, AntennaPod would just send json for EpisodeActions. This would make all of this way easier, could potentially reduce error risks and would also make it easier to use gpoddersync as sync system for other apps.

`array_shift()` is a good recommendation, thanks.